utilities/include/vec4.hpp

```diff
@@ -59,15 +58,14 @@ class alignas(ALIGN_WIDTH) vec4{
                 return sum(temp);
             }
         }
-        
-        CONSTEXPR_CMATH Utype norm() const{
+        constexpr Utype norm() const{
             return std::sqrt(dot(*this,*this));
         }
 
-        CONSTEXPR_CMATH Utype angle(const vec4& v1,const vec4& v2){
+        constexpr Utype angle(const vec4& v1,const vec4& v2){
             return std::acos(dot(v1,v2)/(v1.norm()*v2.norm()));
         }
-        CONSTEXPR_CMATH vec4 unit() const{
+        constexpr vec4 unit() const{
```

Comment:
Why did you change these functions from the `CONSTEXPR_CMATH` macro to `constexpr`?

As of C++20, the `<cmath>` functions are not `constexpr` and hence cannot be used in a constant expression context. However, GCC implements the `<cmath>` functions as `constexpr` in their implementation of the standard library as a non-standard extension. That's why the `CONSTEXPR_CMATH` macro is necessary.